src/main/java/org/snomed/snowstorm/core/data/services/MultiSearchService.java

```diff
@@ -75,23 +63,43 @@ public class MultiSearchService implements CommitListener {
 	
 	private final Logger logger = LoggerFactory.getLogger(getClass());
 	
-	Map<String, String> publishedBranches = new HashMap<>();
-	
-	BoolQueryBuilder cachedBranchesQuery = null;
-	LocalDate cacheDate = null;
+	private final Map<String, String> publishedBranches = new HashMap<>();
+
+	private MultiBranchCriteria cachedBranchCriteria = null;
+	private LocalDate cacheDate = null;
 
 	public Page<Description> findDescriptions(DescriptionCriteria criteria, PageRequest pageRequest) {
+
+		MultiBranchCriteria cachedBranchesQuery = getBranchesQuery();
 		
-		SearchHits<Description> searchHits = findDescriptionsHelper(criteria, pageRequest);
+		MultiBranchCriteria branchesQuery = new MultiBranchCriteria("all-released-plus-requested",cachedBranchesQuery.getTimepoint(),new ArrayList<>(cachedBranchesQuery.getBranchCriteria()));
+		if (criteria.getIncludeBranches().getBranches() != null) {
```

Comment:
@RichardAWood, this looks like it'll throw a NPE for the existing Get endpoint. Because the logic is shared, `includeBranches` will be null when sending a Get request.